sdk/typescript/test/unit/utils/iota-types.test.ts

```
@@ -0,0 +1,56 @@
+// Copyright (c) Mysten Labs, Inc.
+// Modifications Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { describe, expect, it } from 'vitest';
+import { normalizeIotaAddress } from '../../../src/utils/iota-types';
+
+describe('normalizeIotaAddress', () => {
+    it('should convert address to lowercase', () => {
+        expect(normalizeIotaAddress('0xABC123')).toBe(
+            '0x0000000000000000000000000000000000000000000000000000000000abc123',
+        );
+    });
+
+    it('should remove whitespaces', () => {
+        expect(normalizeIotaAddress(' 0xab  c1 23 ')).toBe(
```

Accepting such strings doesn't look like a good idea to me, if it has spaces in between then it's probably some other value and not an address and we should return an error to be save

---

if we go with my suggestion, lets update all these tests so they consider validation on/off

---

I'd say that handling spaces in addresses can be useful, but adding a strict validation should be an optional behavior rather than a default restriction. We introduced the validate flag in the function, providing the following benefits:
1. Preserves existing functionality without breaking previous behavior.
2. Enables output validation directly within the function, eliminating the need for extra checks like isValidIotaAddress.